Fix some compiler warnings

AtariDreams opened the pull request:
Warnings about using the wrong type, wrong types being passed to printf-like functions, etc.

daztucker commented on authfile.c:
```diff
@@ -442,9 +442,7 @@ sshkey_in_file(struct sshkey *key, const char *filename, int strict_type,
-	int r;
-
-	r = ssh_krl_file_contains_key(revoked_keys_file, key);
+	int r = ssh_krl_file_contains_key(revoked_keys_file, key);
```
style guide (https://man.openbsd.org/style.9) says not to use function calls as variable initializers.

daztucker commented on authfile.c:
```diff
@@ -474,10 +472,10 @@ sshkey_check_revoked(struct sshkey *key, const char *revoked_keys_file)
 	int quoted = 0;
 
-	for (; *cp && (quoted || (*cp != ' ' && *cp != '\t')); cp++) {
+	for (cp = *cpp; *cp && (quoted || (*cp != ' ' && *cp != '\t')); cp++) {
```
what warning does this fix?

daztucker commented on openbsd-compat/base64.c:
```diff
@@ -144,8 +144,8 @@ b64_ntop(u_char const *src, size_t srclength, char *target, size_t targsize)
 		srclength -= 3;
 
 		output[0] = input[0] >> 2;
-		output[1] = ((input[0] & 0x03) << 4) + (input[1] >> 4);
```
what's warning is this supposed to fix?

daztucker replied:
You didn't answer my question.

AtariDreams replied:
> Will roll back. Meant for another PR

daztucker reviewed (commented):
There are unrelated changes in the PR, eg changes to the autoconf files.

AtariDreams commented:
@daztucker Addressed!

AtariDreams commented:
> There are unrelated changes in the PR, eg changes to the autoconf files.

The restrict addition is so that __restrict does nothing on compilers that do not support it, while having the restrict keyword on platforms that do.

AtariDreams commented:
> There are unrelated changes in the PR, eg changes to the autoconf files.

I removed the bzero check since I replaced it with memset and then reran autoreconf

daztucker commented on openbsd-compat/bsd-misc.c:
```diff
@@ -387,14 +387,6 @@ llabs(long long j)
-	(void)memset(b, 0, n);
-}
-#endif
-
```
bzero is still used in some places (eg the compat library, netcat in the tests) so this will break on platforms that don't have bzero.  It's also unrelated to the purported purpose of this commit so doesn't belong in this commit (in addition to being pointless churn).

daztucker commented on regress/netcat.c:
```diff
@@ -495,7 +495,7 @@ main(int argc, char *argv[])
 				else {
 					sv = getservbyport(
-					    ntohs(atoi(portlist[i])),
+					    ntohs((int)strtol(portlist[i], NULL, 10)),
```
what warning is this fixing?  atoi already returns an int which is what's needed.

daztucker commented on regress/unittests/test_helper/test_helper.c:
```diff
@@ -392,7 +392,7 @@ tohex(const void *_s, size_t l)
 	assert(r != NULL);
 	for (i = j = 0; i < l; i++) {
-		r[j++] = hex[(s[i] >> 4) & 0xf];
+		r[j++] = hex[s[i] >> 4];
```
what warning is this fixing?

it's also probably wrong on platforms that don't have an integral 8-bit type (Cray is the only example I know of and those are pretty rare these days, but it's allowed by the standards).

daztucker commented on sc25519.c:
```diff
@@ -166,7 +166,7 @@ int sc25519_lt_vartime(const sc25519 *x, const sc25519 *y)
 void sc25519_add(sc25519 *r, const sc25519 *x, const sc25519 *y)
 {
-  int i, carry;
+  crypto_uint32 i, carry;
```
What warning is this fixing?

This code is taken entirely from its upstream (in supercop) and is only ever used for small integer loops so we'd need a really good reason to want to change it.

daztucker reviewed (changes requested):
You've still got unrelated changes in this PR (all of the changes to config.guess, config.sub, install-sh).  You're also mixing different classes of changes in a single commit that are  (eg replacing bzero, which is completely unjustified by the stated goal of this PR).

AtariDreams commented:
Done @daztucker

AtariDreams commented:
Ok, it is all ready now!

AtariDreams commented:
@daztucker Thoughts on this?
